scripts/check_ci_test_suites_to_run.py

```diff
@@ -0,0 +1,689 @@
+) -> dict[str, List[str]]:
+    """Gets the test suites to module mapping from a directory.
+
+    Args:
+        module_mapping_directory: str. The directory to get the
+            test suites to module mapping from.
+        test_suites: list(dict). The list of test suites.
+        main_directory: str | None. The main test type directory.
+            Defaults to None.
+
+    Returns:
+        dict. The test suites to module mapping.
+    """
+    if main_directory is None:
+        main_directory = module_mapping_directory
+    test_suites_to_modules_mapping: dict[str, List[str]] = {}
+    for root, directories, files in os.walk(module_mapping_directory):
+        for file in files:
+            test_suites_to_modules_mapping.update(
+                get_test_suites_to_module_mapping_from_file(
+                    root, file, test_suites, main_directory
+                )
+            )
+        for directory in directories:
+            sub_directory_path = os.path.join(root, directory)
+            test_suites_to_modules_mapping.update(
+                get_test_suites_to_module_mapping_from_directory(
+                    sub_directory_path, test_suites, main_directory
+                )
+            )
+
+    return test_suites_to_modules_mapping
+
+
+def extend_test_suites_without_duplicates(
+    test_suites: List[GenericTestSuiteDict],
+    test_suites_to_extend: List[GenericTestSuiteDict]
+) -> List[GenericTestSuiteDict]:
+    """Extends a list of test suites without duplicates.
+
+    Args:
+        test_suites: list(dict). The list of test suites to extend.
+        test_suites_to_extend: list(dict). The list of test suites to
+            extend with.
+
+    Returns:
+        list(dict). The list of test suites extended without duplicates.
+    """
+    for test_suite in test_suites_to_extend:
+        if test_suite not in test_suites:
+            test_suites.append(test_suite)
+    return test_suites
+
+
+def get_test_suites_affected_by_root_file(
+    root_file: str,
+    test_suites_to_module_mapping: dict[str, List[str]],
+    test_suites: Sequence[GenericTestSuiteDict]
+) -> List[GenericTestSuiteDict]:
+    """Gets the test suites affected by a root file.
+
+    Args:
+        root_file: str. The root file.
+        test_suites_to_module_mapping: dict. The test suites to module mapping
+            for the test type.
+        test_suites: list(dict). The list of test suites.
+
+    Returns:
+        list(dict). The test suites affected by the root file.
+    """
+    test_suites_affected: List[GenericTestSuiteDict] = []
+    for test_suite in test_suites:
+        if test_suite['name'] not in test_suites_to_module_mapping:
+            test_suites_affected.append(test_suite)
+    for (
+        test_suite_name,
+        test_suite_modules
+    ) in test_suites_to_module_mapping.items():
+        test_suite_by_name = get_test_suite_by_name_from_list(
+            test_suites,
+            test_suite_name
+        )
```

Comment:
```suggestion
        test_suite_by_name = get_test_suite_by_name_from_list(
            test_suites, test_suite_name
        )
```

Reply:
In general, when multiple args can fit one line put them on one line.

Reply:
Done.